### PR TITLE
Adding task 5

### PR DESCRIPTION
**Pull request by rsdecoder**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -82,4 +82,32 @@ describe("GET", () => {
...
   });
+  describe('/api/articles', () => {
+    test('STATUS 200, should return an array of article objects', () => {
+        return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then((response) => {
+          const {articles} = response.body;
+          expect(typeof articles).toBe('object')
+          expect(articles[0]['created_at']).toBe('2020-11-03T09:12:00.000Z')
```
Comment (ghost) [inline __tests__/app.test.js:93]: So you're only checking the first article here, what kind of method could you use to check every object has the correct keys?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -82,4 +82,32 @@ describe("GET", () => {
...
+          expect(articles[0]).toHaveProperty('author');
+          expect(articles[0]).toHaveProperty('topic');
+          expect(articles[0]).toHaveProperty('created_at');
+          expect(articles[0]).toHaveProperty('votes');
+          expect(articles[0]).toHaveProperty('article_img_url');
+          expect(articles[0]).toHaveProperty('comment_count');
+          expect(articles[0]).not.toHaveProperty('body')
+        })
+    });
+    test('STATUS - 404, should respond with a 404',() => {
```
Comment (ghost) [inline __tests__/app.test.js:105]: Valid test, but not the right place for it, could you put this outside of the articles describe block.

File: models/articles.models.js [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+        return Promise.reject({ status: 404, msg: "article does not exist" });
+      }
+      return result.rows;
+    });
+};
 
+exports.selectAllArticles = () => {
+    
+    const sqlString = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url,
+    COUNT(comments.article_id) AS comment_count
```
Comment (ghost) [inline models/articles.models.js:17]: have a think how you can cast this as a number rather than a string.

**Review by ghost — COMMENTED** [review]
I'd like you to revisit the testing on this one and see how you can prove all articles have the correct key and data type.

File: __tests__/app.test.js (hunk 1 above)
Comment (rsdecoder) [reply __tests__/app.test.js:93]: should have used map. Will fix it.

Comment (rsdecoder) [reply __tests__/app.test.js:93]: forEach actually

File: __tests__/app.test.js (hunk 2 above)
Comment (rsdecoder) [reply __tests__/app.test.js:105]: will move it out

File: models/articles.models.js (hunk 3 above)
Comment (rsdecoder) [reply models/articles.models.js:17]: CAST (COUNT(comments.article_id AS INTEGER) AS comment_count

**Pull request merged**